### PR TITLE
Refactor: URL utils

### PR DESCRIPTION
**Pull request by marinaaisa**
Bug/issue #96841531, if applicable: 

## Summary

Refactor: URL utils

## Dependencies

NA

## Testing

Steps:
1. Assert that everything works as expected

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Added tests
- [x] Ran `npm test`, and it succeeded
- [x] Updated documentation if necessary

### CODE REVIEW

File: src/utils/assets.js [hunk 1]
```diff
@@ -69,7 +69,8 @@ export function pathJoin(parts) {
  * @param {String} url
  * @return {String}
  */
-export function normalizeAssetUrl(url) {
+export function normalizeUrl(rawUrl) {
```
Comment (dobromir-hristov) [inline src/utils/assets.js:72]: Technically it returns a `path` not a url. Maybe we should rename it?

File: src/utils/assets.js [hunk 2]
```diff
@@ -11,7 +11,7 @@
 /**
  * Utility functions for working with Assets
  */
-import { baseUrl } from 'docc-render/utils/theme-settings';
+const { baseUrl } = window;
```
Comment (dobromir-hristov) [inline src/utils/assets.js:14]: Why the change?

Comment (marinaaisa) [reply src/utils/assets.js:14]: Because it gives a "Dependency cycle via docc-render/utils/url-helper" ESLint error.

The reason is because `assets` imports `theme-settings`, `theme-settings` imports `url-helper` and `url-helper` imports `assets` ♻️
I stopped the cycle by not importing `theme-settings` into `assets`

**Comment by marinaaisa** [comment]
Hey @dobromir-hristov you can give it another look!

**Review by dobromir-hristov — APPROVED** [review]

File: tests/unit/utils/data.spec.js [hunk 3]
```diff
@@ -189,12 +189,11 @@ describe('fetchDataForRouteEnter', () => {
   });
 
   it('calls `fetchData` with a configurable base url', async () => {
-    mockBaseUrl.mockReturnValueOnce('/base-prefix/');
```
Comment (mportiz08) [inline tests/unit/utils/data.spec.js:192]: Why was this removed? It seems like it changes the meaning of this particular test.

Comment (marinaaisa) [reply tests/unit/utils/data.spec.js:192]: I fixed it mocking the `getAbsoluteUrl` function:
https://github.com/apple/swift-docc-render/pull/596/commits/7435a41d891fc28d7aecb48f0a7a5208d0850d0b

Comment (dobromir-hristov) [reply tests/unit/utils/data.spec.js:192]: couldnt we have used the real implementation?

**Comment by marinaaisa** [comment]
@swift-ci test

File: tests/unit/utils/data.spec.js (hunk 3 above)
Comment (marinaaisa) [reply tests/unit/utils/data.spec.js:192]: I couldn't make it work assigning `window.baseUrl` as you mentioned because now adding the baseUrl happens in the `normalizePath` function which it's a different scope, but if you have a suggestion of how to improve this test, please go ahead! :)

**Comment by marinaaisa** [comment]
@swift-ci test

**Comment by marinaaisa** [comment]
@swift-ci test

**Comment by marinaaisa** [comment]
@swift-ci test
